src/components/Image/index.tsx

```diff
@@ -37,29 +37,59 @@ function Image({source: propsSource, isAuthTokenRequired = false, session, onLoa
         },
         [onLoad, updateAspectRatio],
     );
+
+    /**
+     * trying to figure out if the current session is expired or fresh from a necessary reauthentication
+     */
+    const previousSessionAge = useRef<number | undefined>();
+    const validSessionAge: number | undefined = useMemo(() => {
+        if (session?.creationDate) {
+            if (previousSessionAge.current) {
+                if (Math.abs(previousSessionAge.current - session.creationDate) < 60000) {
+                    return session.creationDate;
+                }
```

Comment:
Can you explain why we have this condition?

Reply:
we expect a reauthenticate to happen in less than 60s if the current session was expired. I have made the tests. so the new valid session will be newer than 60s after